### PR TITLE
fix: Organisation can't have a new Github integration when had a prior one deleted

### PR DESCRIPTION
**Pull request by novakzaballa**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

- Add Unique constraints in the GitHub Configuration model.
- Change OneToOneField to ForeignKey in the GitHub Configuration model.
- Add a validation so that an organization does not have more than one valid installation.
- Solve an issue when updating the feature value

## How did you test this code?

- Unit tests added.

### CODE REVIEW

File: api/tests/unit/integrations/github/test_unit_github_views.py [hunk 1]
```diff
@@ -52,6 +52,62 @@ def test_create_github_configuration(
...
 
+@pytest.mark.parametrize(
+    "client",
+    [lazy_fixture("admin_master_api_key_client"), lazy_fixture("admin_client")],
+)
+def test_cannot_create_github_configuration_due_to_unique_constraint(
+    client: APIClient,
+    organisation: Organisation,
+    github_configuration: GithubConfiguration,
+) -> None:
```
Comment (zachaysan) [inline api/tests/unit/integrations/github/test_unit_github_views.py]:
```suggestion
def test_cannot_create_github_configuration_due_to_unique_constraint(
    admin_client_new: APIClient,
    organisation: Organisation,
    github_configuration: GithubConfiguration,
) -> None:
```

File: api/tests/unit/integrations/github/test_unit_github_views.py [hunk 2]
```diff
@@ -52,6 +52,62 @@ def test_create_github_configuration(
...
+
+@pytest.mark.parametrize(
+    "client",
+    [lazy_fixture("admin_master_api_key_client"), lazy_fixture("admin_client")],
+)
+def test_cannot_create_github_configuration_when_the_organization_already_has_an_integration(
+    client: APIClient,
+    organisation: Organisation,
+    github_configuration: GithubConfiguration,
+) -> None:
```
Comment (zachaysan) [inline api/tests/unit/integrations/github/test_unit_github_views.py]:
```suggestion
def test_cannot_create_github_configuration_when_the_organization_already_has_an_integration(
    admin_client_new: APIClient,
    organisation: Organisation,
    github_configuration: GithubConfiguration,
) -> None:
```

File: api/integrations/github/views.py [hunk 3]
```diff
@@ -61,6 +62,17 @@ def get_queryset(self):
...
+    def create(self, request, *args, **kwargs):
+
+        organisation_id = self.kwargs["organisation_pk"]
+        if GithubConfiguration.has_github_configuration(organisation_id):
+            raise DuplicateGitHubIntegration
+
+        try:
+            return super().create(request, *args, **kwargs)
+        except IntegrityError:
+            raise DuplicateGitHubIntegration
```
Comment (zachaysan) [inline api/integrations/github/views.py]: Is there a way of looking at the `IntegrityError` message in order to raise `DuplicateGitHubIntegration` only when the preconditions have been met?

**Review by zachaysan — COMMENTED** [review]
Overall looks pretty good, just have a couple questions.

File: api/tests/unit/integrations/github/test_unit_github_views.py (hunk 2 above)
Comment (novakzaballa) [reply api/tests/unit/integrations/github/test_unit_github_views.py]: Done.

File: api/tests/unit/integrations/github/test_unit_github_views.py (hunk 1 above)
Comment (novakzaballa) [reply api/tests/unit/integrations/github/test_unit_github_views.py]: Done.

File: api/integrations/github/views.py (hunk 3 above)
Comment (novakzaballa) [reply api/integrations/github/views.py]: Done

**Review by zachaysan — APPROVED** [review]
Looks good buddy :)

**Pull request merged**